sources/packages/backend/apps/api/src/route-controllers/program-year/_tests_/e2e/program-year.students.controller.getProgramYears.e2e-spec.ts

```diff
@@ -0,0 +1,157 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { ProgramYear } from "@sims/sims-db";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getStudentToken,
+  FakeStudentUsersTypes,
+} from "../../../../testHelpers";
+
+describe("ProgramYearStudentsController(e2e)-getProgramYears", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    appDataSource = dataSource;
+    app = nestApplication;
+  });
+
+  it("Should get the list of the available program years.", async () => {
+    // Arrange
+    const endpoint = "/students/program-year/options-list";
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const programYearList = response.body;
+        expect(programYearList).toHaveLength(3);
+        expect(programYearList).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              id: 1,
+              description:
+                "(2021-2022) - Study starting between August 01, 2021 and July 31, 2022",
+            }),
+            expect.objectContaining({
+              id: 2,
+              description:
+                "(2022-2023) - Study starting between August 01, 2022 and July 31, 2023",
+            }),
+            expect.objectContaining({
+              id: 3,
+              description:
+                "(2023-2024) - Study starting between August 01, 2023 and July 31, 2024",
+            }),
+          ]),
+        );
+      });
+  });
+
+  it("Should not include inactive program years in the list.", async () => {
+    // Arrange
+    const programYearRepo = appDataSource.getRepository(ProgramYear);
+    const programYear = await programYearRepo.findOneBy({ id: 2 });
+    programYear.active = false;
+    await programYearRepo.save(programYear);
+    const endpoint = "/students/program-year/options-list";
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
```

Comment:
This failure is not in test scenario.

![image](https://user-images.githubusercontent.com/54600590/225688244-861092eb-c660-4e11-afbe-92a7d9fe1d9b.png)

Reply:
I didn't see a test scenario for this. I believe it is because of the program year created in data seed right? We cannot achieve a not found scenario

Reply:
How can I test that scenario if the DB migration already includes 3 program years? As discussed earlier we should not remove those program years (https://github.com/bcgov/SIMS/pull/1803#discussion_r1137839064). Also those program years are used for some tests and some applications use them.

Reply:
Yes that is what exactly I mentioned that the program years are already there.